engine/executor/dm/worker.go

```diff
@@ -164,33 +164,29 @@ func (w *dmWorker) Workload() model.RescUnit {
 	return 0
 }
 
-// OnMasterFailover implements lib.WorkerImpl.OnMasterFailover
-func (w *dmWorker) OnMasterFailover(reason framework.MasterFailoverReason) error {
-	w.Logger().Info("dmworker.OnMasterFailover")
-	return nil
-}
-
 // OnMasterMessage implements lib.WorkerImpl.OnMasterMessage
 func (w *dmWorker) OnMasterMessage(ctx context.Context, topic p2p.Topic, message p2p.MessageValue) error {
 	w.Logger().Info("dmworker.OnMasterMessage", zap.String("topic", topic), zap.Any("message", message))
 	return nil
 }
 
 // CloseImpl implements lib.WorkerImpl.CloseImpl
-func (w *dmWorker) CloseImpl(ctx context.Context) error {
+func (w *dmWorker) CloseImpl(ctx context.Context) {
 	w.Logger().Info("close the dm worker", zap.String("task-id", w.taskID))
-	var recordErr error
 	// unregister jobmaster client
-	if err := w.messageAgent.UpdateClient(w.masterID, nil); err != nil {
-		w.Logger().Error("failed to update message client", zap.Error(err))
-		recordErr = err
+	if w.messageAgent != nil {
```

Comment:
Why not early return if `w.messageAgent == nil` to avoid nil check everywhere? See: L185.

Reply:
the old logic is 
1. delete message client to jobmaster
2. stop DM units
3. close message agent
 
Can we change the order of them? @GMHDBJD

Reply:
yes, you can change the order